sys/crypto/psa_riot_cipher/chacha20.c

```diff
@@ -92,32 +94,40 @@ psa_status_t psa_cipher_chacha20_update(psa_cipher_chacha20_ctx_t *ctx,
     DEBUG("RIOT ChaCha20 Cipher update");
     psa_status_t status = PSA_ERROR_CORRUPTION_DETECTED;
 
-    if (output_size < input_length)
+    if (output_size < (((ctx->buffer_length + input_length) >> 6) << 6))
     {
         return PSA_ERROR_BUFFER_TOO_SMALL;
     }
 
-    if (!ctx->is_finished)
+    size_t input_index = 0;
+    size_t output_index = 0;
+    // check if buffer contains data
+    if (ctx->buffer_length)
     {
```

Comment:
```suggestion
    if (ctx->buffer_length > 0) {
```

maybe slightly more expressive